### ISSUE TITLE
22.07 - logger.hpp:14:44: error: format not a string literal and no format arguments

### ISSUE THREAD

**Issue by martinkg** [DISTRHO/Cardinal#325]
Hi,

get the following error when compiling cardinal-22.07 on Fedora 36:

```
Compiling unless_modules/src/avoider.cpp
g++ unless_modules/src/avoider.cpp -Wall -Wextra -pipe -MD -MP -g -fno-gnu-unique -fPIC -DPIC -DNDEBUG -O2 -ffast-math -fdata-sections -ffunction-sections -fvisibility=hidden -DHAVE_FFTW3F -DARCH_LIN -DBUILDING_PLUGIN_MODULES -I../dpf/dgl/src/nanovg -I../dpf/distrho -I../include -I../include/osdialog-stub -I../include/simd-compat -DHAVE_X11 -DCARDINAL_SYSDEPS  -I../deps/sysroot/include -I../src -I../src/Rack/include -I../src/Rack/include/dsp -I../src/Rack/dep/filesystem/include -I../src/Rack/dep/glfw/include -I../src/Rack/dep/nanosvg/src -I../src/Rack/dep/oui-blendish -I../src/Rack/dep/pffft -pthread -DHAVE_SNDFILE -Wno-unused-parameter -Wno-unused-variable -Wno-deprecated-declarations -std=gnu++11 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fvisibility-inlines-hidden -fvisibility-inlines-hidden -fexceptions -DDISTRHO_NAMESPACE=CardinalDISTRHO -DDGL_NAMESPACE=CardinalDGL -DDGL_USE_NANOVG_FBO -fno-finite-math-only -fno-strict-aliasing  -faligned-new -Wno-abi -U_GLIBCXX_ASSERTIONS -Wp,-U_GLIBCXX_ASSERTIONS -c -o ../build/plugins/unless_modules/src/avoider.cpp.o \
	-DSelection=unless_modulesSelection -DmodelSelection=modelunless_modulesSelection -DSelectionWidget=unless_modulesSelectionWidget \
	-DpluginInstance=pluginInstance__unless_modules
annobin: unless_modules/src/avoider.cpp: Warning: -D_GLIBCXX_ASSERTIONS not defined
In file included from ../src/Rack/include/common.hpp:269,
                 from ../include/common.hpp:31,
                 from ../src/Rack/include/rack.hpp:22,
                 from ../include/rack.hpp:20,
                 from unless_modules/src/unless.hpp:1,
                 from unless_modules/src/avoider.cpp:1:
unless_modules/src/_arth.hpp: In static member function 'static arth::ArtData arth::ArtData::load(std::string)':
../src/Rack/include/logger.hpp:14:44: error: format not a string literal and no format arguments [-Werror=format-security]
   14 | #define INFO(format, ...) rack::logger::log(rack::logger::INFO_LEVEL, __FILE__, __LINE__, __FUNCTION__, format, ##__VA_ARGS__)
      |                           ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
unless_modules/src/_arth.hpp:516:7: note: in expansion of macro 'INFO'
  516 |       INFO(("Loaded ART " + slug + ".art").c_str());
      |       ^~~~
cc1plus: some warnings being treated as errors
```
installed is:
gcc-Version 12.1.1 20220507 (Red Hat 12.1.1-1) (GCC)

i use this command to compile:
make -O -j2 V=1 VERBOSE=1 SKIP_STRIPPING=true NOOPT=true V=1 SYSDEPS=true

**Comment by falkTX** [comment]
Issue comes from Rack, the same macro is used whether the caller uses any arguments or not.
Best to build without `-Werror=format-security`, this is what makes the build fail.

Alternatively I can tweak the build system so it injects `-Wno-format-security` after the CLI build flags.
Tell me which way you prefer.

**Comment by martinkg** [comment]
I prefer the second method, thanks for your feedback.

**Comment by falkTX** [comment]
Should be fixed in 7c40addf345b231fba3779f61c305058594595f2
Testing would be appreciated.

**Comment by martinkg** [comment]
> Should be fixed in [7c40add](https://github.com/DISTRHO/Cardinal/commit/7c40addf345b231fba3779f61c305058594595f2) Testing would be appreciated.

compiles without problems until this .... error, but this is a other problem.

`Compiling override/Scene.cpp
g++ override/Scene.cpp -Wall -Wextra -pipe -MD -MP -g -fno-gnu-unique -fPIC -DPIC -DNDEBUG -O2 -ffast-math -fdata-sections -ffunction-sections -fvisibility=hidden -DARCH_LIN -DPRIVATE= -I../dpf/dgl/src/nanovg -I../dpf/distrho -I../include -I../include/simd-compat -IRack/include -DCARDINAL_SYSDEPS  -IRack/dep/filesystem/include -IRack/dep/fuzzysearchdatabase/src -IRack/dep/glfw/include -IRack/dep/nanosvg/src -IRack/dep/osdialog -IRack/dep/oui-blendish -IRack/dep/pffft -DHAVE_LIBLO  -pthread -Wno-unused-but-set-variable -Wno-unused-parameter -Wno-unused-result -Wno-unused-variable -std=gnu++11 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fvisibility-inlines-hidden -fvisibility-inlines-hidden -fexceptions -DDISTRHO_NAMESPACE=CardinalDISTRHO -DDGL_NAMESPACE=CardinalDGL -DDGL_USE_NANOVG_FBO -fno-finite-math-only -fno-strict-aliasing -faligned-new -Wno-abi -DnsvgParseFromFile=nsvgParseFromFileCardinal -U_GLIBCXX_ASSERTIONS -Wp,-U_GLIBCXX_ASSERTIONS -Wno-format-security -c -o ../build/rack/override/Scene.cpp.o
gcc-annobin: override/Scene.cpp: Warning: -D_GLIBCXX_ASSERTIONS not defined
override/Scene.cpp: In member function 'virtual void rack::app::ResizeHandle::onEnter(const rack::widget::Widget::EnterEvent&)':
override/Scene.cpp:112:74: error: 'GLFW_RESIZE_NWSE_CURSOR' was not declared in this scope; did you mean 'GLFW_HRESIZE_CURSOR'?
  112 |                 glfwSetCursor(APP->window->win, glfwCreateStandardCursor(GLFW_RESIZE_NWSE_CURSOR));
      |                                                                          ^~~~~~~~~~~~~~~~~~~~~~~
      |                                                                          GLFW_HRESIZE_CURSOR
make[1]: *** [Makefile:258: ../build/rack/override/Scene.cpp.o] Error 1
`
